include/setup.php

```diff
@@ -317,9 +317,19 @@
 // Loads English localized strings by default (must go before config.php)
 require 'languages/english.php';
 
+// Try to support one WebSVN-installation hosting multiple different SVNParentPaths, distinguished
+// by their location. Per location the web server needs to set some environment variable providing
+// the path to the config to include.
+//
+// https://stackoverflow.com/questions/3050444/when-setting-environment-variables-in-apache-rewriterule-directives-what-causes
+$envConfPath = preg_grep('/^(?:REDIRECT_)*WEBSVN_PATH_CONF$/', array_keys($_SERVER));
```

Comment:
I don't understand why this is necessary, why not just say `$_SERVER['WEBSVN_PATH_CONF']`?

Reply:
Have a look at the SO-link, environment variable names change in case redirects are used at least for `httpd`. This happened to me and in a setup like mine using redirects seems pretty common, to add things like `index.php` if not provided etc. So I already dealt with those problems.

Reply:
Can you provide a reference for that expect that 10 year old SO question. Does this `mod_php` or `mod_rewrite`?

Reply:
Not sure who exactly is doing that, most likely `mod_rewrite` or something along that lines, but not `mod_php`. The latest comments in the highest voted answer are only very few years old and this thing simply happened to me yesterday while I implemented the changes.

> FWIW, here are some Apache docs that explicitly talk about the "REDIRECT_" prefix: [httpd.apache.org/docs/2.0/custom-error.html](http://httpd.apache.org/docs/2.0/custom-error.html). I sympathize with how annoying it is, though — just got bitten by it, myself. – jwd Jun 15 '16 at 16:39

https://stackoverflow.com/questions/3050444/when-setting-environment-variables-in-apache-rewriterule-directives-what-causes#comment63143106_10128290

> Apache HTTPD Subversion history indicates this code in modules/http/http_request.c that runs rename_original_env() was added in the bulk import from 1.3.x CVS, and I can't seem to find that earlier history anywhere now. You should file a bug report to have them explain and document this more clearly. – Josip Rodin Mar 29 '17 at 12:30

https://stackoverflow.com/questions/3050444/when-setting-environment-variables-in-apache-rewriterule-directives-what-causes#comment73268141_10128290

> I am running with Apache 2.4 + mod_fastcgi + php-fpm with 'alias' directive to mapping to the external fastcgi path, I think this is the cause of my 'REDIRECT_' env var renaming, after switching to using 'mod_proxy_fastcgi', even with same .htaccess rule, the problem no longer exists. – weynhamz Dec 14 '17 at 4:19

https://stackoverflow.com/questions/3050444/when-setting-environment-variables-in-apache-rewriterule-directives-what-causes#comment82573020_10128290

Reply:
Obviously, no one ever took a look at the code: https://github.com/apache/httpd/blob/69a49ab530eaa18dfc46fd1967022d0f77245e27/modules/http/http_request.c#L528-L543

https://httpd.apache.org/docs/2.4/custom-error.html contains some `REDIRECT_`, but I must admit that is is poor or not documented at all.

Reply:
> Obviously, no one ever took a look at the code:

The first two comments I quoted are dealing with exactly what your are providing, only in older versions.

So what do you expect me to do? The link about `custom-error` doesn't say anything that environment variables are replaced by `REDIRECTED_`-ones, it only talks about adding some special ones and doesn't even focus on rewrites in general. So in my opinion it's not better than to link to SO. Adding the bug might be worth it, adding the lines in the code OTOH not, as `http_request` is already mentioned in the SO-comments if people case that much.

Reply:
The custom-error docs are nonsense. You might want to add a link to the BZ issue. That's the best we have.